### DISCUSSION TITLE
How I save the original filename when use upscaler?

### DISCUSSION THREAD

**Discussion by roflosaur** [WASasquatch/was-node-suite-comfyui#267] [Q&A] [answered]
How I can keep the original filename when use upscaler? I use this for mass texture upscaling, and have serious problems with saving file sequence after setting index. Here is a way to set original filenames after upscale?
![1](https://github.com/WASasquatch/was-node-suite-comfyui/assets/144611248/95489f24-fcda-4fe2-8711-0286bf7e887f)

**Comment by WASasquatch** [comment]
The IDX works in single image mode, incremental just steps through each image in a loop each prompt run. So to use IDX you could use a Number Counter as a input for the index set to number of images expected to increment over. 

On the Image Save node you'd want to use overwrite_mode prefix_as_filename.

**Comment by x5nder** [comment]
@roflosaur did you make this work? Can you share a screenshot of how you did it? I also want to upscale a huge batch of photos while retaining the file names (should be possible) and JPG metadata (don't think this is possible), any help is welcome!

**Comment by alirezasadg** [comment, marked as answer]
right click on (image save) and select convert widget to input then connect filenaem_text
i prefer to put overwrite mode true

**Reply by roflosaur** [reply]
Thanks! Now my life is much easier.
